Add device trigger translation documentation (#734)
home-assistant/developers.home-assistant: docs/device_automation_trigger.md

```

The return value is a function that detaches the trigger.

#### Add text and translations

> The *New Automation* user interface will display a human readable string in the device automation mapped to the event
> type.  Update `strings.json` and if you used script.scaffold then `translations/en.json` with the full set of supported event types:
> 

Reviewer: "New" is relative in time, so better to avoid that 😉 

```suggestion
The Automation user interface will display a human-readable string in the device automation mapped to the event
type.  Update `strings.json` and if you used `script.scaffold` then `translations/en.json` with the full set of supported event types:

```

> type.  Update `strings.json` and if you used `script.scaffold` then `translations/en.json` with the full set of supported event types:
> 
> ```json
> {
>    "device_automation": {
>     "trigger_type": {
>       "water_detected": "Water detected",
>       "noise_detected": "Noise detected"
>     }
> }

Reviewer 2: ```suggestion
type.  Update `strings.json` with the trigger types and subtypes that you support:

```json
{
   "device_automation": {
    "trigger_type": {
      "remote_button_short_press": "\"{subtype}\" button pressed",
      "remote_button_short_release": "\"{subtype}\" button released"
    },
    "trigger_subtype": {
      "turn_on": "Turn on",
      "turn_off": "Turn off"
    }
}
\```

To test your translations during development, run `python3 -m script.translations develop`.

```

Reviewer 2: Comment failure because of the JSON block being inside the suggestion
